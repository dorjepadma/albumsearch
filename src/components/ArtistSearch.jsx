import React from 'react';
import PropTypes from 'prop-types';

const ArtistSearch = ({ artistSearch, handleChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type='text' 
      value={artistSearch} 
      onChange={handleChange} />
    <button>Find Groovy Music</button>
  </form>
);

ArtistSearch.propTypes = {
  handleChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  artistSearch: PropTypes.string,
};

export default ArtistSearch;
