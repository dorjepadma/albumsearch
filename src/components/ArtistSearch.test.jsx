import React from 'react';
import { shallow } from 'enzyme';
import ArtistSearch from './ArtistSearch';

describe('Controls component', () => {
  it('renders App', () => {
    const wrapper = shallow(<ArtistSearch text="Prince" onChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
