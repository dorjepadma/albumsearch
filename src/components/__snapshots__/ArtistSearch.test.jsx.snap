// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders App 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button>
    Find Groovy Music
  </button>
</form>
`;
